lib/components/molecules/AtomicCarousel/AtomicCarousel.js

```diff
@@ -0,0 +1,341 @@
+// @flow
+/**
+ *
+ * AtomicCarousel
+ *
+ */
+import React, { useState, useEffect, useCallback } from 'react';
+import styled from 'styled-components';
+import classnames from 'classnames';
+
+import styles from './AtomicCarousel.style';
+import type { Props } from './types';
+import { ListVanilla } from '../List';
+import Button from '../../atoms/Button';
+import Icon from '../../atoms/Icon';
+import Heading from '../../atoms/Heading';
+import useInterval from './useInterval.hook';
+
+const AtomicCarousel = ({
+  dots,
+  navigation,
+  children,
+  className,
+  perSlideFrame,
+  prevLabel,
+  nextLabel,
+  headers,
+  heading,
+  displayHeader,
+  slideTransitionTime,
+  autoPlay,
+  fade,
+  playIcon,
+  nextIcon,
+  prevIcon,
+  pauseIcon,
+}: Props): Node => {
+  const [currentSlide, setCurrentSlide] = useState(1);
+  const [data, setData] = useState([]);
+  const [slideCount, setSlideCount] = useState(0);
+  const [isAnimationRunning, setAnimation] = useState(false);
+  const [mouseStatus, setMouseStatus] = useState(false);
+
+  const specificNode = useCallback((index, callback) => callback(index));
+  const handlePrev = useCallback(() => {
+    setCurrentSlide(currentSlide - 1);
+  });
+  const handleNext = useCallback(() => {
+    setCurrentSlide(currentSlide + 1);
+  });
+  const handleAnimation = useCallback(() => {
+    setAnimation(!isAnimationRunning);
+    setMouseStatus(!isAnimationRunning);
+  });
+
+  useEffect(() => {
+    if (children !== undefined) {
+      let i = 0;
+      let slideItems = [];
+      if (children !== undefined && !Array.isArray(children)) {
```

Comment:
Why you checking for children's existence so many times. Look at parent condition.